src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -1209,13 +1240,21 @@ private void EvaluateItemGroupElement(ProjectItemGroupElement itemGroupElement)
             }
 #endif
 
-            bool itemGroupConditionResult = EvaluateCondition(itemGroupElement, ExpanderOptions.ExpandPropertiesAndItems, ParserOptions.AllowPropertiesAndItemLists);
+            bool itemGroupConditionResult;
+            if (lazyEvaluator != null)
+            {
+                itemGroupConditionResult = lazyEvaluator.EvaluateConditionWithCurrentState(itemGroupElement, ExpanderOptions.ExpandPropertiesAndItems, ParserOptions.AllowPropertiesAndItemLists);
```

Comment:
>EvaluateConditionWithCurrentState [](start = 57, length = 33)

I saw that item definition groups expand some metadata when evaluating their conditions. Didn't look farther into it, whether it's some actual item metadata or some special item definition metadata